connect/src/com/telenor/connect/ui/ConnectWebViewClient.java

```diff
@@ -94,6 +109,70 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
     }
 
     @Override
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+        // Only implement interception logic when we have both cellular and WiFi interfaces active
+        if (ConnectSdk.getCellularNetwork() == null || ConnectSdk.getWifiNetwork() == null) {
+            return null;
+        }
+        if (shouldFetchThroughCellular(request.getUrl().toString())) {
+            return fetchUrlThroghCellular(request.getUrl().toString());
+        }
+        return null;
+    }
+
+    public boolean shouldFetchThroughCellular(String url) {
+        try {
+            String host = (new URL(url)).getHost();
+            String hostIp = InetAddress.getByName(host).getHostAddress();
+            return ConnectSdk
+                    .getWellKnownConfig()
+                    .getNetworkAuthenticationTargetIps()
+                    .contains(hostIp);
+        } catch (MalformedURLException|UnknownHostException e) {
+            return false;
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public WebResourceResponse fetchUrlThroghCellular(String originalUrl) {
+        String newUrl = originalUrl;
+        int attempts = 0;
+        do {
+            Network interfaceToUse = ConnectSdk.getCellularNetwork();
+            HttpURLConnection connection = null;
+            InputStream inputStream = null;
+            try {
```

Comment:
Yes. All of them except getContentType & getContentEncoding throw so I think it's more readable.